src/other/utils.jl

```diff
@@ -178,3 +178,133 @@ function _nt_like_hash(v, h::UInt)
 
     return xor(objectid(Tuple(propertynames(v))), h)
 end
+
+_findall(B) = findall(B)
+
+@inline _blsr(x)= x & (x-1) 
+const _msk64 = ~UInt64(0)
+
+# findall optimized for B::BitVector
+# the main idea from Base.findall(B::BitArray)
+function _findall(B::BitVector)::Union{UnitRange{Int}, Vector{Int}}
+    nnzB = count(B)
+    nnzB == 0 && return Int[]
+    nnzB == length(B) && return 1:length(B)
+    I = Vector{Int}()
+    Bc = B.chunks
+    Bi = 1 # block index
+    i1 = 1 # index of current block beginng in B
+    i = 1  # index of the _next_ one in I
+    c = Bc[1] # current block
+
+    start = -1 # the begining of ones block
+    stop = -1  # the end of ones block
+    
+    @inbounds while true # I not materialized
+        i > nnzB && return start:start + i - 2 # all ones in B found
+
+        if c == 0
+            if start != -1 && stop == -1
+                stop = i1 - 1
+            end
+            while c == 0 # no need to return here as we returned above
+                i1 += 64
+                Bi += 1
+                c = Bc[Bi]
```

Comment:
are you sure this is always `@inbounds`?

Reply:
ah - you are sure as if we went out of bounds we would return in line 204 - right?

Reply:
Exactly, if we already found all trues, we leave in 204, otherwise we will find some trues later, not in this loop.